ports/views.py

```diff
@@ -98,34 +99,45 @@ def portdetail(request, name):
 
 
 def all_builds_view(request):
-    filter_applied = False
-    if request.method == 'POST':
-        if request.POST['status-filter']:
-            filter_by = request.POST['status-filter']
-            if filter_by == "All Builds":
-                all_builds = BuildHistory.objects.all().order_by('-time_start')
-            else:
-                all_builds = BuildHistory.objects.filter(status=filter_by).order_by('-time_start')
-                filter_applied = filter_by
-        else:
-            return HttpResponse("Something went wrong")
-    else:
-        all_builds = BuildHistory.objects.all().order_by('-time_start')
-    paginated_builds = Paginator(all_builds, 100)
-    page = request.GET.get('page', 1)
-    try:
-        builds = paginated_builds.get_page(page)
-    except PageNotAnInteger:
-        builds = paginated_builds.get_page(1)
-    except EmptyPage:
-        builds = paginated_builds.get_page(paginated_builds.num_pages)
+    builders = Builder.objects.all().values_list('name', flat=True)
 
     return render(request, 'ports/all_builds.html', {
-        'all_builds': builds,
-        'filter_applied': filter_applied,
+        'builders': builders,
     })
 
 
+def all_builds_filter(request):
+    if request.method == 'POST':
+        builder = request.POST['builder']
+        status = request.POST['status']
+        page = request.POST['page']
+
+        if builder == "All Builders" and status == "All Builds":
+            builds = BuildHistory.objects.all().order_by('-time_start')
+        elif builder == "All Builders" and status != "All Builds":
+            builds = BuildHistory.objects.filter(status=status).order_by('-time_start')
+        elif builder != "All Builders" and status == "All Builds":
+            builds = BuildHistory.objects.filter(builder_name__name=builder).order_by('-time_start')
+        else:
+            builds = BuildHistory.objects.filter(status=status, builder_name__name=builder).order_by('-time_start')
```

Comment:
I'm not super happy about how this is done. Imagine that you need to apply 10 different filters. Will you really write 1024 if-else sentences? I'm not 100% sure about the correct syntax in Django to achieve that, but this could probably be done with a single `BuildHistory.objects.filter(...)` call with some generalization of the filter calls.

Reply:
Thank you. I have now switched to using django-filters and this is no longer needed. To maintain consistency, I will switch all the filters to use django-filters.